### DISCUSSION TITLE
How to compile zip.js so that it can adapt to lower versions of browsers, such as chrome 49

### DISCUSSION THREAD

**Discussion by huangminsky** [gildas-lormeau/zip.js#285] [Q&A]
How to compile zip.js so that it can adapt to lower versions of browsers, such as chrome 49 ?

【rollup.config.js】
```js
import resolve from '@rollup/plugin-node-resolve';
import babel from "rollup-plugin-babel";
export default [
    {
        input: 'index',
        output: {
            name: 'timeout',
            file: 'bundle.js',
            format: 'cjs'
        },
        plugins: [
            babel({ runtimeHelpers: true})
        ],
        // plugins: [resolve(), babel({ babelHelpers: 'bundled' })]
    }
];
```
【.babelrc】
```json
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "modules": false,
        "useBuiltIns": "usage",
        "corejs": "3.19.1",
        "targets": {
          "chrome": "49"
        }
      }
    ]
  ],
  "plugins": [
    [
      "@babel/plugin-transform-runtime"
    ]
  ],
  "ignore": [
    "node_modules/**"
  ]
}
```

------------------------------------
Then I imported bundle.js, and zipWriter.add(file.name, new zip.BlobReader(file), options);

However, an error is reported in the lower version browser
![image](https://user-images.githubusercontent.com/6838399/141061407-3c07a293-9422-4f0d-80de-5d266f4fecbb.png)
Then I check the error log ,It's from z-worker-inline.js
![image](https://user-images.githubusercontent.com/6838399/141061311-260a0c11-099c-4da3-a085-926f6bcb849d.png)

anyone know what happened ??

**Comment by gildas-lormeau** [comment]
I think the issue is related to the fact that the code used for the web worker is not transpiled with babel. If this is the case, I think calling `zip.configure({ useWebWorkers: false });` should help to circumvent the issue.
To fix it, you would need to include a plugin for babel in the rollup script I use to build zip.js, see https://github.com/gildas-lormeau/zip.js/blob/f432082fcf208b012edddd3bd99242e078215fd5/rollup.config.js#L4-L9.

**Reply by huangminsky** [reply]
Thank you for saving me a lot of time

**Reply by gildas-lormeau** [reply]
You're welcome. Feel free to ping me if you need more details.

**Comment by gildas-lormeau** [comment]
For the record, zip.js now includes a rollup script to build files for older browsers like IE 11, see https://github.com/gildas-lormeau/zip.js/blob/master/rollup-es5.config.js.

**Comment by huangminsky** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。
